### PR TITLE
Verbosity in acceptance tests should not be forced

### PR DESCRIPTION
**Pull request by radeksimko**
This condition has been added along with first batch of acceptance tests, so it's not something brought for a reason that would appear during test runs in different environment and by different people I reckon.

So I'm raising the question: Is there any point in forcing users to make acceptance tests verbose?

You may argue we're not forcing them, but that's what such error message effectively does as it's added to each acceptance test case in **non-verbose** mode.

IMO user will see all fails/errors even in normal mode without `-v` and if needed, he'll dig in and pass `-v` to find out more.

### CODE REVIEW

**Comment by mitchellh** [comment]
The reason they're verbose is because if they're not, you don't see what test is running and the progress of that test. Some tests take dozens of minutes and its more helpful to see that work is being done than it is to have cleaner output. 

Therefore the "he'll dig in and pass -v" is not the best scenario because if a test takes 10 minutes to fail, that is pretty annoying.

So we decided to trade up front verbosity for a better developer experience for test failures. Does this make sense?

**Comment by radeksimko** [comment]
Keep in mind that many people may not (and in reality won't) have all the access keys for all cloud services in environment variables, because e.g. they may only want to run AWS tests.

In that case, all tests will be very fast and you'll see a long verbose output (even though you didn't ask for it) and you're forced to filter out all of these first

> Acceptance tests must be run with the -v flag on tests

to find out what the real problem is.

I agree that the progress reporting is a problem, but I'm not sure this is the best way of dealing with it.
One example of dealing with it could be something like ascii progress bar which would tell you _"yeah, something is happening, I'm now busy with this test"_.

Generally it's more about the end result you see after all tests finish and the tool should be able to make it a little bit more interactive (e.g. printing "Running test aws => package x", "Running test aws => package y" but only leave "ok test aws" and remove all the verbosity if user doesn't want it using escape sequences).

You may see that sort of functionality in most packaging managers on most distros... user knows that there's something happening, but all that "verbosity" doesn't stay on the screen... it does not report every single percentage as a new line.

So to sum up, it's more of a problem of go testing package I reckon and I think your reasoning supports the need for such feature as well.

**Comment by radeksimko** [comment]
Someone on the `#go-nuts` IRC channel brought a valid point, even those are acceptance tests, a single test probably shouldn't take dozens of minutes.

Do we really have to create a giant stack and test all dependencies on that? Can't we just cut it into smaller pieces and test dependencies (or whatever reason of the test is) between resources separately?

A single API call AFAIK does not take too much time, the problem is the amount of resources we create within a single test.

**Comment by mitchellh** [comment]
I think the difference that `#go-nuts` probably isn't grasping is that these are just acceptance tests. We happen to use the Go test framework for it, but they're not at all unit tests. They're meant to black box test the system end-to-end for a feature, and that in the case of Terraform is inevitably slow.

However, it isn't the _only_ tests. We do defense in depth: testing layers multiple times at multiple locations. So we have full unit test coverage of all our internals and also some of our providers, but we also do acceptance tests. This allows us to find bugs at multiple levels.

Note that you can run unit tests without verbosity, because they fail fast and there are literally thousands of them. But the acceptance tests are usually more targeted, are much slower due to the inherent nature of the black box testing, and are run much less frequently as a result.

The reason we create many resources within a single test is because to test an EC2 instance in a VPC you need a VPC, but then you need a subnet, but then you need a routing table, etc. Sometimes the provider forces our hand.

I'd say each acceptance test does test the minimal amount it can, its just slow. :) 

I hope this answers everything adequately and you can understand our testing methodology more: defense in depth, acceptance tests are the final layer. Given that, I would prefer to keep verbosity on!
